packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3973,11 +3995,49 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       : TextPosition(offset: textBoundary.getLeadingTextBoundaryAt(caretOffset) ?? 0);
   }
 
+  /// Select a paragraph around the given position.
+  ///
+  /// A paragraph boundary is defined as the range that most closely
+  /// encapsulates the given position within two line terminators. If
+  /// a line terminator does not exist in a given direction the selection
+  /// extends to the start/end of the document in that direction.
+  void selectParagraph({ required Offset position, SelectionChangedCause? cause }) {
+    selectParagraphsInRange(from: position, cause: cause);
+  }
+
+  /// Selects the set of paragraphs in a document that intersect a given range of global positions.
+  ///
+  /// The set of paragraphs selected are not strictly bounded by the range of global positions.
+  ///
+  /// The first and last endpoints of the selection will always be at the beginning and end of a
+  /// paragraph respectively.
+  void selectParagraphsInRange({ required Offset from, Offset? to, SelectionChangedCause? cause }) {
+    final TextBoundary paragraphBoundary = ParagraphBoundary(_value.text);
+    final int fromPoint = renderEditable.getPositionForPoint(from).offset;
+    int? toPoint = to == null ? null : renderEditable.getPositionForPoint(to).offset;
```

Comment:
Maybe merge this line and the next one to a single assignment?

Reply:
I mostly did that to avoid calling `.getPositionForPoint` twice, but I moved some of the code around.